proposed/0013-immutable-data-naming/0013-immutable-data-naming.md

```diff
@@ -13,75 +13,91 @@ This RFC outlines the system components and design for how the three immutable d
 
-Normal : normal_name = hash(im.content), handled by DM(normal_name), 4 copies on the pmid_nodes picked up by that group (one copy on each node)
+The Normal DMs (DataManagers comprising closest group to the Normal name), forward client Put requests to the Backup and Sacrificial DMs, and forward Get requests to the Backup and Sacrificial DMs only if required (e.g. in case of heavy churn where the Normal copies are unavailable).  This minimises client exposure to the network data types and their management.
 
-Backup : backup_name = hash(normal_name), handled by DM(backup_name), 4 copies on the pmid_nodes picked up by that group (one copy on each node)
+## Implementation Details
 
-Sacrificial : sacrificial_name = hash(backup_name), handled by DM(sacrificial_name), 4 copies on the pmid_nodes picked up by that group (one copy on each node)
+### Put
 
-MaidManagers will charge the client issuing the put request with 4 copies of the data, as the backup and sacrificial copies are allowed to be removed from the SAFE network according to the network status.
+MMs (MaidManagers) should respond with failure for any Put or Get requests from Clients for Backup or Sacrificial types.
 
-The portal DMs (DataManagers that being the closest group to narmal_name), fowards the put/get requests to the second/third order DMs (i.e. DataManagers closing to backup_name and sacrificial_name). This reduces the exposure of network data types to client side to minimum.
+DMs handling a Put request for a Normal chunk from MM should:
 
-## Implementation Details
+* Send Put to the two PM (PmidManager) groups closest to the data name
+* Construct a Backup and a Sacrificial copy of the chunk and send Put requests for these to the DM groups for each
 
+DMs handling a Put request for a Backup or Sacrificial chunk from DM should:
 
-## Planned Work
+* Send Put to the two PM (PmidManager) groups closest to the data name
 
-1. Vault
-    1. DataManager
-    1. MaidManager
-    1. put flow refactoring
-    1. get flow refactoring
+PNs (PmidNodes) without enough free space to handle a Put request for any type chunk from PMs should respond with failure.
 
-1. Routing
-    1. ImmutableData sanity check
+DMs handling a Put failure for Normal or Backup chunks should send Delete requests for Sacrificial chunks which the same PN holds to the PM group.  Once the responses have all arrived, the initial Put request should be retried.
 
-1. Client
-    1. Put
-    1. Get
+If a PN is ultimately unable to store a chunk even after removing all Sacrificial chunks, it should be marked as "bad" by the DMs and not retried as a holder for that chunk.  The Put process stops at a DM group once a copy is held on two different PNs from the close group for that chunk, or once all Vaults in the close group have been attempted and failed.
```

Comment:
I think the later on chapters says PN is not going to remove Sacrificial chunks?